gateway/src/apicast/gc.lua

```diff
@@ -0,0 +1,53 @@
+-- In LuaJIT and Lua 5.1, the __gc metamethod does not work in tables, it only
+-- works in "userdata". This module introduces a workaround to make it work
+-- with tables.
+
+local getmetatable = getmetatable
+local newproxy = newproxy
+local ipairs = ipairs
+local pairs = pairs
+local tostring = tostring
+
+local _M = {}
+
+local function __gc(proxy)
+  local t = getmetatable(proxy).__table
+  local mt = getmetatable(t)
+
+  if mt and mt.__gc then mt.__gc(t) end
+end
+
+local function __call(proxy, ...)
+  local t = getmetatable(proxy).__table
+  local mt = getmetatable(t)
+
+  if mt and mt.__call then
+    return mt.__call(...)
+  end
+end
+
+function _M.enable(table)
```

Comment:
We could rename this library to `setmetatable_gc(table, mt)`. 

It could make more sense when using the `__metatable` key. 
All changes to GC'd objects metatables should be done with this method.